manifests/target.pp

```diff
@@ -43,41 +59,43 @@
 #   from
 #
 #   * Set to ``['ALL']`` to allow from any location
+#   * If empty, will disable the use of pam_access for this user
 #
 # @param user_max_logins
 #   The ``pam_limits`` restricting the number of concurrent sessions permitted for
 #   ``username``
 #
+#   If set to ``undef``, will not restrict the maximum number of logins for the
+#   user
+#
 class simp_bolt::target (
-  Boolean                    $enable_user                  = false,
-  String                     $user_name                    = 'simp_bolt',
+  Boolean                    $create_user                  = true,
+  String[1]                  $user_name                    = 'simp_bolt',
   Array[String[1]]           $disallowed_users             = ['root'],
   Optional[String[8]]        $user_password                = undef,
   Stdlib::Unixpath           $user_home                    = "/var/local/${user_name}",
-  Integer                    $user_uid                     = 1779,
-  Integer                    $user_gid                     = $user_uid,
+  Integer[500]               $user_uid                     = 1779,
+  Integer[500]               $user_gid                     = $user_uid,
   Optional[Array[String[1]]] $user_ssh_authorized_keys     = undef,
   String[1]                  $user_ssh_authorized_key_type = 'ssh-rsa',
-  String                     $user_sudo_user               = 'root',
+  Optional[String[1]]        $user_sudo_user               = 'root',
   Boolean                    $user_sudo_password_required  = false,
-  Array[String]              $user_sudo_commands           = ['ALL'],
-  Array[String]              $user_allowed_from            = [pick(fact('puppet_server'), '127.0.0.1')],
-  Integer                    $user_max_logins              = 1
+  Array[String[1],1]         $user_sudo_commands           = ['ALL'],
+  Array[String[1]]           $user_allowed_from            = [pick(fact('puppet_server'), 'LOCAL')],
```

Comment:
I don't think LOCAL access should be permitted unless the system is also a Bolt controller.  There is no for a local account on the system to access this account, which will presumably have sudo access to root without a password.